### ISSUE TITLE
Predicate pruning test fails when using "new grouping" branch

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#2153]
__Steps to reproduce:__
Update datafusion to point at the https://github.com/alamb/arrow-datafusion/tree/alamb/fixup_scalar branch in my fork of datafusion that contains a subset of the changes to support null in group by in https://github.com/apache/arrow-datafusion/pull/793

__Expected behaviour:__
All IOx tests should pass

__Actual behaviour:__
`sql_predicate_pushdown_correctness_13 ` fails in the following way (aka when scanning data from parquet all rows are filtered, incorrectly)

```shell

---- sql::sql_predicate_pushdown_correctness_13 stdout ----
Running scenario '--------------------- Data in read buffer'
SQL: '"SELECT * from restaurant where system > 5.0 and system < 7.0 and town = 'reading'"'
Running scenario '--------------------- Data in object store only '
SQL: '"SELECT * from restaurant where system > 5.0 and system < 7.0 and town = 'reading'"'
thread 'sql::sql_predicate_pushdown_correctness_13' panicked at 'assertion failed: `(left == right)`
  left: `["+-------+--------+--------------------------------+---------+", "| count | system | time                           | town    |", "+-------+--------+--------------------------------+---------+", "| 632   | 6      | 1970-01-01T00:00:00.000000130Z | reading |", "+-------+--------+--------------------------------+---------+"]`,
 right: `["++", "++"]`: 

expected:

[
    "+-------+--------+--------------------------------+---------+",
    "| count | system | time                           | town    |",
    "+-------+--------+--------------------------------+---------+",
    "| 632   | 6      | 1970-01-01T00:00:00.000000130Z | reading |",
    "+-------+--------+--------------------------------+---------+",
]
actual:

[
    "++",
    "++",
]

', query_tests/src/sql.rs:709:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I am fairly sure this is related to a bug in the pruning logic that is exposed now that `ScalarValue` handles NULL values correctly.

I found this while testing a fix for something else in https://github.com/influxdata/influxdb_iox/pull/2151

**Comment by alamb** [comment]
Here is a PR that shows the problem https://github.com/influxdata/influxdb_iox/pull/2154

**Comment by alamb** [comment]
I believe I have found at least one problem in parquet statistics generation: https://github.com/apache/arrow-rs/issues/641

The new code in https://github.com/apache/arrow-datafusion/pull/793 effectively enables parquet pruning for tag columns (previously they were ignored by the pruning logic, as they were dictionaries and not supported)

I have also confirmed if we disable parquet predicate pushdown (can leave IOx's pruning) we do get the correct answer. 

Thus, my plan for this ticket is likely to disable parquet predicate pruning and file a ticket to re-enable it when the upstream arrow bug is fixed which will unblock fixing the usize issue. Gah!
